components/net/image_cache.rs

```diff
@@ -384,29 +381,35 @@ impl ImageCache for ImageCacheImpl {
             store: Arc::new(Mutex::new(ImageCacheStore {
                 pending_loads: AllPendingLoads::new(),
                 completed_loads: HashMap::new(),
-                placeholder_image: get_placeholder_image(&webrender_api, &rippy_data).ok(),
+                placeholder_image: get_placeholder_image(&webrender_api, &rippy_data),
                 placeholder_url: ServoUrl::parse("chrome://resources/rippy.png").unwrap(),
                 webrender_api: webrender_api,
             })),
         }
     }
 
-    /// Return any available metadata or image for the given URL,
-    /// or an indication that the image is not yet available if it is in progress,
-    /// or else reserve a slot in the cache for the URL if the consumer can request images.
-    fn find_image_or_metadata(
+    fn get_image(&self, url: &ServoUrl) -> Option<Arc<Image>> {
+        let store = self.store.lock().unwrap();
+        match store.get_completed_image_if_available(url, UsePlaceholder::No) {
+            Some(Ok(ImageOrMetadataAvailable::ImageAvailable(x, _))) => Some(x),
+            _ => None,
+        }
+    }
+
+    fn track_image(
         &self,
         url: ServoUrl,
         use_placeholder: UsePlaceholder,
         can_request: CanRequestImages,
-    ) -> Result<ImageOrMetadataAvailable, ImageState> {
-        debug!("Find image or metadata for {}", url);
+    ) -> ImageCacheResult {
         let mut store = self.store.lock().unwrap();
+
         if let Some(result) = store.get_completed_image_if_available(&url, use_placeholder) {
             debug!("{} is available", url);
-            return result;
+            return ImageCacheResult::Available(result.unwrap());
         }
 
+        let (sender, _receiver) = ipc::channel().unwrap();
```

Comment:
This looks suspicious to me - I believe callers of track_image should be passing a sender as an argument so they can make use of the receiver. Right now any notifications go into the void because the receiver is ignored completely.

Reply:
not sure what I was thinking here!